contracts/beacon/Voucher.sol

```diff
@@ -164,11 +184,69 @@ contract Voucher is OwnableUpgradeable, IVoucher {
             workerpoolOrder,
             requestOrder
         );
-        $._sponsoredAmounts[dealId] = sponsoredAmount;
+        $._voucherMatchedDeals[dealId].exists = true;
+        $._voucherMatchedDeals[dealId].sponsoredAmount = uint64(sponsoredAmount);
         emit OrdersBoostMatchedWithVoucher(dealId);
         return dealId;
     }
 
+    function claim(bytes32 dealId, uint256 index) external {
```

Comment:
Having separate claim/refund methods for classic and boost will consume less gas (and will also keep things consistent with other parts [matchOrders & matchOrdersBoost in voucher & claims methods in poco])